server/service/apple_mdm.go

```diff
@@ -5531,45 +5552,68 @@ func replaceFleetVarInItem(ctx context.Context, ds fleet.Datastore, target *cmdT
 			hardwareSerial, ok := caVarsCache[string(fleet.FleetVarHostHardwareSerial)]
 			if !ok {
 				var err error
-				hardwareSerial, ok, err = getHostHardwareSerial(ctx, ds, target, hostUUID)
+				hostLite, ok, err = hydrateHost(ctx, ds, target, hostLite)
```

Comment:
Would love to be able to combine this call, with the call in profile_verifier (Windows), and put it alongside the others that were moved to `profile_variables`, that way we can share the logic of fetching a hardware serial from the host, take a look at how they do in terms of reporting errors and calling onError function so each platform can respectively handle it the way they want.